### PR TITLE
Add Config.env_prefix option

### PR DESCRIPTION
**Pull request by alex-oleshkevich**
This PR adds `Config.env_prefix` argument.

```python
import os
from starlette.config import Config

os.environ['APP_DEBUG'] = 'yes'
os.environ['ENVIRONMENT'] = 'dev'

config = Config(env_prefix='APP_')

DEBUG = config('DEBUG') # lookups APP_DEBUG, returns "yes"
ENVIRONMENT = config('ENVIRONMENT') # lookups APP_ENVIRONMENT, raises KeyError as variable is not defined
```

### CODE REVIEW

**Review by Kludex — COMMENTED** [review]
@alex-oleshkevich If you don't mind, would you be able to give a bit more about the motivation here?

This is a pretty straightforward feature, but no one ever asked before.

**Comment by alex-oleshkevich** [comment]
This is a common practice to avoid naming conflicts.
E.g. there is `ENV` variable defined by the OS, and the app itself may require `ENV` like "dev" or "prod". So you can prefix your envvars to separate app's ones and third-party.

This situation pretty common in non-docker environments. Or, it may happen in docker-compose when 2 services require the same variable but with different value.

These are basically two examples I recall so far.

File: docs/config.md [hunk 1]
```diff
@@ -113,6 +113,25 @@ from starlette.config import environ
 environ['TESTING'] = 'TRUE'
 ```
 
+## Reading prefixed environment variables
+
+You can namespace the environment variables by setting `env_prefix` argument.
+
+**myproject/settings.py**:
+
+```python
```
Comment (Kludex) [inline docs/config.md]:
```suggestion
```python title="myproject/settings.py"
```
Can't we do this?

**Review by Kludex — APPROVED** [review]
Ok, let's go. :shrug:

**Comment by Kludex** [comment]
Please check if the documentation changes that I suggested work, it may need a plugin. 👀

**Comment by alex-oleshkevich** [comment]
> Please check if the documentation changes that I suggested work, it may need a plugin. eyes

it looks good 
image

**Pull request merged**